src/lib_c5t_popen2.cc

```diff
@@ -141,9 +141,9 @@ void C5T_POPEN2(std::vector<std::string> const& cmdline,
     return std::thread(
         [](std::function<void(const std::string)> cb_line, int read_fd, int terminate_fd) {
           struct pollfd fds[2];
-          fds[0].fd = read_fd;
```

Comment:
Here the tweak works since I now check the termination signal first.

(Strictly speaking, there was no need to flip them in the args to `::poll()`, just swapping the `if`-s would do.)

My best guess is that the true reason this code did not work before on OS X is that once the child process is terminated, `::poll()` notifies about its `stdout` as well.

I did not explore much further, just made sure to give priority to the check that the `::waitpid()` has already succeeded.

Tested afresh with https://github.com/dimacurrentai/exec_on_macos/.